extensions/sql/lease/src/main/java/org/eclipse/dataspaceconnector/sql/lease/SqlLeaseContextConfiguration.java

```diff
@@ -0,0 +1,68 @@
+/*
+ *  Copyright (c) 2020 - 2022 Microsoft Corporation
+ *
+ *  This program and the accompanying materials are made available under the
+ *  terms of the Apache License, Version 2.0 which is available at
+ *  https://www.apache.org/licenses/LICENSE-2.0
+ *
+ *  SPDX-License-Identifier: Apache-2.0
+ *
+ *  Contributors:
+ *       Microsoft Corporation - initial API and implementation
+ *
+ */
+
+package org.eclipse.dataspaceconnector.sql.lease;
+
+import org.eclipse.dataspaceconnector.spi.transaction.TransactionContext;
+
+import java.sql.Connection;
+import java.time.Duration;
+import java.util.Objects;
+
+/**
+ * Utility class to dynamically assemble a {@link SqlLeaseContext} and execute operations on it using a {@link Connection}.
+ */
+public class SqlLeaseContextConfiguration {
```

Comment:
Isn't this basically a builder? I think it would be cleaner to split the "withConnection()" up into a "connection()" that sets the connection and a "build()" that builds.

Reply:
it's very similar, but not quite. Users should not store references to the `SqlLeaseContext`, because connections should not be kept open. I know that it is still possible, but less obvious at least. The idea is that users configure the `SqlLeaseContextConfiguration` once, and then call `withConnection(...).[acquire|break]` several times. So the intent here was not to create a builder, that sets several potentially optional fields. I'll update the source comments.